docs/source/examples/scaling.md

```diff
@@ -0,0 +1,161 @@
+    print(
+        "{: >12}  {: ^14} {: ^14} {: ^15}".format(
+            scale_factor, len(demo), folded_depth, id_insert_depth
+        )
+    )
+```
+
+As expected we have $\mathtt{depth} * \mathtt{scale\_factor} = \mathtt{scaled\_depth}$.
+
+## Using noise scaling methods
+
+Here we demo how you can use these noise-scaling technique in ZNE.
+First we define an [executor](https://mitiq.readthedocs.io/en/stable/guide/executors.html) which is needed to tell Mitiq how to run the circuit.
+Here we choose a simple density matrix simulation with depolarizing noise which acts between every circuit layer.
+
+```{code-cell} ipython3
+from mitiq.zne import execute_with_zne
+import cirq
+
+def execute(circuit, noise_level=0.05):
+    noisy_circuit = circuit.with_noise(cirq.depolarize(p=noise_level))
+    return (
+        cirq.DensityMatrixSimulator()
+        .simulate(noisy_circuit)
+        .final_density_matrix[0, 0]
+        .real
+    )
+```
+
+We can then pass the desired noise-scaling method into `execute_with_zne` using the `scale_noise` keyword argument.
+
+```{code-cell} ipython3
+execute_with_zne(generate_ghz_circuit(6), execute, scale_noise=insert_id_layers)
+```
+
+To give a slightly more systematic understanding of the differences between these two methods, we will perform a small experiment on the following variational circuit.
+
+```{code-cell} ipython3
+def variational_circuit(gamma):
+    q0, q1 = cirq.LineQubit.range(2)
+
+    return cirq.Circuit(
+        [
+            cirq.rx(gamma)(q0),
+            cirq.CNOT(q0, q1),
+            cirq.rx(gamma)(q1),
+            cirq.CNOT(q0, q1),
+            cirq.rx(gamma)(q0),
+        ]
+    )
+```
+
+We can run this circuit many times, varying $\gamma$ each time, and compute ideal and noisy expectation values to compare to the mitigated versions.
+We do this comparison by using an improvement factor (IF) which is defined as
+```{math}
+\left|\frac{\text{ideal} - \text{noisy}}{\text{ideal} - \text{mitigated}}\right|.
+```
+
+```{code-cell} ipython3
+from random import uniform
+import numpy as np
+
+results = {"fold": [], "id": []}
+for _ in range(100):
+    gamma = uniform(0, 2 * np.pi)
+    circuit = variational_circuit(gamma)
+
+    ideal_expval = execute(circuit, noise_level=0.0)
+    noisy_expval = execute(circuit)
+    folded_expval = execute_with_zne(circuit, execute, scale_noise=fold_global)
+    id_expval = execute_with_zne(circuit, execute, scale_noise=insert_id_layers)
+
+    noisy_error = abs(ideal_expval - noisy_expval)
+    folded_IF = noisy_error / abs(ideal_expval - folded_expval)
+    scaled_IF = noisy_error / abs(ideal_expval - id_expval)
+
+    results["fold"].append(folded_IF)
+    results["id"].append(scaled_IF)
+
+print("Avg improvement factor (`fold_global`):      ", round(np.average(results["fold"]), 4))
+print("Avg improvement factor (`insert_id_layers`): ", round(np.average(results["id"]), 4))
+```
+
+As we can see, both techniques offer an improvement.
```

Comment:
Any idea why noise scaling improvement is 6x greater than folding? Under what conditions would we expect one to perform (much) better than the other?

Reply:
I was also very surprised by this! It's a good sign, but to be truthful, I'm not really sure why it's performing so well! My _guess_ is that is has something to do with applying inverse gates via folding cancels out some of the noise amplification? tagging @andreamari to see if he's got any ideas.

Reply:
You can also use `mitiq.zne.inference.Factory.plot_data` and `mitiq.zne.inference.Factory.plot_fit` to see the scaling effects.

Reply:
> I was also very surprised by this! It's a good sign, but to be truthful, I'm not really sure why it's performing so well! My _guess_ is that is has something to do with applying inverse gates via folding cancels out some of the noise amplification? tagging @andreamari to see if he's got any ideas.

I think it strongly depends on the simulated noise model.
The Cirq instruction `circuit.with_noise(cirq.depolarize(p=noise_level))` adds noise to **each qubit** after **each moment** independently from the applied gates. For this type of noise model, the insertion of identity layers is  optimal.

On the other hand, for a noise model in which a noise channel is applied after each gate, unitary folding is probably better.

Reply:
thanks @andreamari! 

@natestemen, WDYT about this as a brief explanation at the end?

```suggestion
Both noise scaling techniques offer an improvement, but for this particular problem identity insertion outperforms unitary folding. 
The noise model adds noise to each qubit after each moment independently from the applied gates. For this type of noise model, the insertion of identity layers is optimal.

On the other hand, for a noise model in which a noise channel is applied after each gate, unitary folding would likely outperform identity insertion.
```

Reply:
I used the suggestion as a basis for [`4de60f9`](https://github.com/unitaryfund/mitiq/pull/1633/commits/4de60f91a51bd0419c221a2a092912a2604bac0b). WDYT?